SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/DynamicActorAwakeAfterDormantChangePropertyTest.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "DynamicActorAwakeAfterDormantChangePropertyTest.h"
+
+#include "DormancyTestActor.h"
+
+// This test checks whether changes on a dormant actor are replicated when the actor becomes awake.
+
+ADynamicActorAwakeAfterDormantChangePropertyTest::ADynamicActorAwakeAfterDormantChangePropertyTest()
+{
+	Author = "Matthew Sandford";
+}
+
+void ADynamicActorAwakeAfterDormantChangePropertyTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Step 1 - Spawn dormancy actor and change NetDormancy to DORM_DormantAll
+	AddStep(TEXT("ServerSpawnDormancyActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AActor* Actor = CreateDormancyTestActor();
+		Actor->SetNetDormancy(DORM_DormantAll);
+		FinishStep();
+	});
+
+	// Step 2 - Client check NetDormancy is DORM_DormantAll
+	AddStep(
+		TEXT("ClientRequireDormancyAndRepProperty"), FWorkerDefinition::AllClients, nullptr, nullptr,
+		[this](float DeltaTime) {
+			RequireDormancyActorCount(1);
+			RequireDormancyAndRepProperty(DORM_DormantAll, 0);
+			FinishStep();
+		},
+		5.0f);
+
+	// Step 3 - Server set TestIntProp to 1
+	AddStep(TEXT("ServerModifyNetDormancy"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		for (TActorIterator<ADormancyTestActor> Iter(GetWorld()); Iter; ++Iter)
+		{
+			ADormancyTestActor* DormancyTestActor = *Iter;
+			DormancyTestActor->TestIntProp = 1;
+		}
+		FinishStep();
+	});
+
+	// Step 4 - Give chance for property to be replicated to clients
+	AddStep(
+		TEXT("ClientWaitForReplication"), FWorkerDefinition::AllClients, nullptr,
+		[this]() {
+			FTimerManager& TimerManager = GetWorld()->GetTimerManager();
+			TimerManager.SetTimer(
```

Comment:
I think it may be simpler to do a `StepTimer` float and then just `RequireGreaterThan(StepTimer, 0.5f, TEXT(""));`

Reply:
Kind like the timer manager. I also made this a lot neater by adding finishstep into the timer lambda.

Reply:
I just don't like the timer manager. But it's fine.

This is an alternative from the spatialTestNetOwnership:
```
	//  All workers check that the number of RPCs received by the authoritative server is correct.
	AddStep(
		TEXT("SpatialTestNetOwnershipAllWorkersTestCount2"), FWorkerDefinition::AllWorkers,
		[this]() -> bool {
			return NetOwnershipCube->GetOwner() == nullptr;
		},
		[this]() {
			TimeInStepHelper = 0.0f;
		},
		[this, TestLocations](float DeltaTime) {
			TimeInStepHelper += DeltaTime;
			RequireCompare_Float(TimeInStepHelper, EComparisonMethod::Greater_Than_Or_Equal_To, 1.0f,
								 TEXT("Have to wait 1 second to make sure the RPC doesn't arrive."));
			RequireEqual_Int(NetOwnershipCube->ReceivedRPCs, TestLocations.Num(),
							 TEXT("RPC sent while not owning the cube should not have been received."));
			FinishStep();
		},
		10.0f);
```

Reply:
Now, this would be nicer to use if we had a public auto-incrementing timer available to all tests